python-sdk/src/astro/databases/snowflake.py

```diff
@@ -475,35 +478,62 @@ def create_table_using_native_schema_autodetection(
             },
         )
 
+    @staticmethod
+    def get_cols_case(cols: tuple[str]):
+        result = []
+        for col in cols:
+            if col.isupper():
+                result.append(1)
+            elif col.islower():
+                result.append(0)
+            else:
+                return "mixed"
+
+        total = sum(result)
+        if total == 0:
+            return "lower"
+        elif total == len(cols):
+            return "upper"
+
+    @staticmethod
+    def use_quotes(cols: tuple[str]):
+        return SnowflakeDatabase.get_cols_case(cols) in ["upper", "mixed"]
+
     def create_table_using_schema_autodetection(
         self,
         table: BaseTable,
         file: File | None = None,
         dataframe: pd.DataFrame | None = None,
-        columns_names_capitalization: ColumnCapitalization = "lower",
+        columns_names_capitalization: ColumnCapitalization = "original",
     ) -> None:
         """
         Create a SQL table, automatically inferring the schema using the given file.
+        Overriding default behaviour and not using the `prep_table` since it doesn't allow the adding quotes.
 
         :param table: The table to be created.
         :param file: File used to infer the new table columns.
         :param dataframe: Dataframe used to infer the new table columns if there is no file
         """
+        if file is None:
+            if dataframe is None:
+                raise ValueError(
+                    "File or Dataframe is required for creating table using schema autodetection"
+                )
+            source_dataframe = dataframe
+        else:
+            source_dataframe = file.export_to_dataframe(nrows=LOAD_TABLE_AUTODETECT_ROWS_COUNT)
 
-        # Snowflake don't expect mixed case col names like - 'Title' or 'Category'
-        # we explicitly convert them to lower case, if not provided by user
-        if columns_names_capitalization not in ["lower", "upper"]:
-            columns_names_capitalization = "lower"
-
-        if file:
-            dataframe = file.export_to_dataframe(
-                nrows=settings.LOAD_TABLE_AUTODETECT_ROWS_COUNT,
-                columns_names_capitalization=columns_names_capitalization,
-            )
-
-        # Snowflake doesn't handle well mixed capitalisation of column name chars
-        # we are handling this more gracefully in a separate PR
-        super().create_table_using_schema_autodetection(table, dataframe=dataframe)
+        pandas_tools.write_pandas(
+            conn=self.hook.get_conn(),
+            df=source_dataframe,
+            table_name=table.name.upper(),
+            schema=table.metadata.schema,
+            database=table.metadata.database,
+            chunk_size=DEFAULT_CHUNK_SIZE,
+            quote_identifiers=SnowflakeDatabase.use_quotes(source_dataframe),
+            auto_create_table=True,
+        )
+        self.truncate_table(table)
```

Comment:
Why do we truncate table after loading the data?

Reply:
So earlier we were using prep_table() which just takes a dataframe and then comes up with an SQL query to create the table and we executed the query. But that function doesn't give us the quotes=True/False. So this was the next best option I had to create a table. Since we are just expecting an empty table to be created, we are truncating. Loading of actual data happens in two separate flows - 1. Pandas 2. Native.

Reply:
So we are double loading the file (although the first time we do -- we probably only load 1000 rows for schema detection)

Reply:
We should add it as comment